experimental/python/databricks/bundles/build.py

```diff
@@ -0,0 +1,479 @@
+
+    return functions, diagnostics
+
+
+def _load_object(qualified_name: str) -> tuple[Optional[Callable], Diagnostics]:
+    diagnostics = Diagnostics()
+    [module_name, name] = qualified_name.split(":")
+
+    common_error = qualified_name == "resources:load_resources"
+
+    try:
+        module = importlib.import_module(module_name)
+    except Exception as exc:
+        if isinstance(exc, ModuleNotFoundError) and exc.name == "resources":
+            if common_error:
+                return None, _explain_common_import_error(exc)
+
+        return None, Diagnostics.from_exception(
+            exc=exc,
+            summary=f"Failed to import module '{module_name}'",
+        )
+
+    try:
+        return getattr(module, name), diagnostics
+    except AttributeError as exc:
+        if common_error:
+            return None, _explain_common_import_error(exc)
+
+        return None, Diagnostics.from_exception(
+            exc=exc,
+            summary=f"Name '{name}' not found in module '{module_name}'",
+            location=Location(file=module.__file__) if module.__file__ else None,
+        )
+    except Exception as exc:
+        return None, Diagnostics.from_exception(
+            exc=exc,
+            summary=f"Failed to load function '{name}' from module '{module_name}'",
+        )
+
+
+def _explain_common_import_error(exc: Exception) -> Diagnostics:
+    # a common case when default name of the module is not found
+    # we can give a hint to the user how to fix it
+    explanation = """Make sure to create a new Python file at resources/__init__.py with contents:
+    
+from databricks.bundles.core import load_resources_from_current_package_module
+
+
+def load_resources():
+    return load_resources_from_current_package_module()
+"""
+    return Diagnostics.from_exception(
+        exc=exc,
+        summary="Can't find function 'load_resources' in module 'resources'",
+        explanation=explanation,
+    )
+
+
+def _load_function(qualified_name: str) -> tuple[Optional[Callable], Diagnostics]:
+    [instance, diagnostics] = _load_object(qualified_name)
+    [module_name, name] = qualified_name.split(":")
+
+    if diagnostics.has_error():
+        return None, diagnostics
+
+    if instance and not isinstance(instance, Callable):
+        return None, Diagnostics.create_error(
+            f"Function '{name}' in module '{module_name}' is not callable",
+        )
+
+    return instance, diagnostics
+
+
+def _load_resource_mutator(
+    qualified_name: str,
+) -> tuple[Optional[ResourceMutator], Diagnostics]:
+    [instance, diagnostics] = _load_object(qualified_name)
+    [module_name, name] = qualified_name.split(":")
+
+    if diagnostics.has_error():
+        return None, diagnostics
+
+    if instance and not isinstance(instance, ResourceMutator):
+        return None, Diagnostics.create_error(
+            f"'{name}' in module '{module_name}' is not instance of ResourceMutator, did you decorate it with @job_mutator?",
```

Comment:
Is this decorator still used?

The mutators as configured in the bundle are a list of functions already, no?

Reply:
We still need decorator to define the type of resources that are handled by this function. It will be relevant when we have more than 1.